eth/sync.go

```diff
@@ -222,7 +222,7 @@ func (cs *chainSyncer) loop() {
 
 		case <-cs.pm.quitSync:
 			if cs.doneCh != nil {
-				cs.pm.downloader.Cancel()
+				cs.pm.downloader.Terminate() // Double term is fine, Cancel would block until queue is emptied
```

Comment:
Terminate calls Cancel too... ?

Reply:
Yes

Reply:
But it closes a quit channel first so that intenral components know they need to die, not just not-restart